src/probnum/diffeq/perturbedsolvers/perturbedstepsolver.py

```diff
@@ -0,0 +1,88 @@
+"""ODE solver as proposed by Abdulle and Garegnani."""
+import numpy as np
+
+from probnum import _randomvariablelist, diffeq, randvars, utils
+from probnum.type import FloatArgType
+
+
+class PerturbedStepSolver(diffeq.ODESolver):
+    """ODE Solver based on Scipy that introduces uncertainty by perturbing the time-
+    steps."""
+
+    def __init__(
+        self,
+        # diffeq.WrappedScipyRungeKutta
+        solver,
+        noise_scale: FloatArgType,
+        perturb_function,
+        random_state=None,
+    ):
+        def perturbation_step(step):
+            return perturb_function(
+                step=step,
+                solver_order=solver.order,
+                noise_scale=noise_scale,
+                random_state=utils.as_random_state(random_state),
```

Comment:
I dont think this should be done. If for instance `random_state` is a seed (i.e. an integer; which shouldnt be done but happens), this perturbation function *always* returns the same perturbation. If instead, you wrap `random_state` through `utils.as_random_state` outside of this function call (i.e. outside of `def perturbation_step`), things work as expected.